test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadReference/stop/stop002.java

```diff
@@ -101,75 +112,196 @@ private int runIt(String args[], PrintStream out) {
             return quitDebuggee();
         }
 
-        ThreadReference thrRef = null;
-        if ((thrRef =
-                debuggee.threadByName(DEBUGGEE_THRNAME)) == null) {
+        ThreadReference thrRef = debuggee.threadByName(DEBUGGEE_THRNAME);
+        if (thrRef == null) {
             log.complain("TEST FAILURE: method Debugee.threadByName() returned null for debuggee thread "
                 + DEBUGGEE_THRNAME);
             tot_res = Consts.TEST_FAILED;
             return quitDebuggee();
         }
 
-        Field doExit = null;
+        Field stopLoop1 = null;
+        Field stopLoop2 = null;
         ObjectReference objRef = null;
+        ObjectReference throwableRef = null;
         try {
             // debuggee main class
-            ReferenceType rType = debuggee.classByName(DEBUGGEE_CLASS);
-
-            suspendAtBP(rType, DEBUGGEE_STOPATLINE);
-            objRef = findObjRef(thrRef, DEBUGGEE_LOCALVAR);
+            mainClass = debuggee.classByName(DEBUGGEE_CLASS);
 
-            // this field is used to indicate that debuggee has to
-            // stop looping
-            doExit = rType.fieldByName(DEBUGGEE_FIELD);
+            suspendAtBP(mainClass, DEBUGGEE_STOPATLINE);
+            objRef = findObjRef(thrRef, DEBUGGEE_NON_THROWABLE_VAR);
+            throwableRef = findObjRef(thrRef, DEBUGGEE_THROWABLE_VAR);
 
-            log.display("Resuming debuggee VM ...");
-            vm.resume();
-            log.display("Resumption of debuggee VM done");
+            // These fields are used to indicate that debuggee has to stop looping
+            stopLoop1 = mainClass.fieldByName(DEBUGGEE_STOP_LOOP1_FIELD);
+            stopLoop2 = mainClass.fieldByName(DEBUGGEE_STOP_LOOP2_FIELD);
+            if (stopLoop1 == null || stopLoop2 == null) {
+                throw new RuntimeException("Failed to find a \"stop loop\" field");
+            }
 
-            log.display("\nTrying to stop debuggee thread \"" + thrRef
-                + "\"\n\tusing non-throwable object reference \""
-                + objRef + "\" as a parameter ...");
+            log.display("non-throwable object: \"" + objRef + "\"");
+            log.display("throwable object:     \"" + throwableRef + "\"");
+            log.display("debuggee thread:      \"" + thrRef + "\"");
 
-// Check the tested assersion
+            /*
+             * Test #1: verify using a non-throwable object with stop() fails appropriately.
+             */
```

Comment:
The same suggestion about refactoring each subtest case into a separate method.
There is some overhead in doing this, so it is up to you but worth to consider.